File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.java
import java.util.* ; 
import java.io.* ;

class Solution{
    public static void main(String[] args) throws IOException{   
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in)) ;
        for(int t = 1 ; t <= 10 ; t++){
        	
            int N = Integer.parseInt(br.readLine()) ;
        	int[] builds =Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray()   ; 

        	int answer = 0; 
        	for(int i = 2 ; i < N - 2 ; i++){
            	int height = builds[i]; 
            	int max = Math.max(Math.max(builds[i-2], builds[i-1]), Math.max(builds[i+1], builds[i+2])); 
            	if(height > max)
                	answer += height - max ; 
        	}
        	System.out.println("#"+t+" "+answer);
		}
    }
}

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
for T in range(1, 11):
    L = int(input())
    buildings = list(map(int, input().split()))
 
    result = 0
    for i in range(2, L-2):
        h = buildings[i] - max(buildings[i-2:i]+buildings[i+1:i+3])
        result += h if h > 0 else 0
 
    print(f'#{T} {result}')